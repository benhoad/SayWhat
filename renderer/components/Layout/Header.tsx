interface IHeaderProps {
  children?: any;
}

export default function Header({ children }: IHeaderProps) {
  return (
    <>
      <div className="Header">{children}</div>

      <style jsx>{`
        .Header {
          height: 2rem;
          flex-grow: 0;
          flex-shrink: 0;
          overflow: hidden;
          display: flex;
          background: var(--color-header-background);
          color: white;
        }

        .Header :global(button) {
          padding: 0 0.5rem;
          background: none;
          border: none;
          color: white;
          font-family: var(--font-family-sans-serif);
          font-size: 0.7rem;
          line-height: 2rem;
          cursor: pointer;
          white-space: nowrap;
        }
        .Header :global(button:hover) {
          background-color: rgba(255, 255, 255, 0.1);
        }

        .Header :global(button[data-icon]) {
          padding-left: 2rem;
          background-repeat: no-repeat;
          background-position: 0.5rem 48%;
          background-size: 1rem 1rem;
        }

        .Header :global(button[data-icon="menu"]) {
          background-image: url(./images/MenuButton.svg);
        }

        .Header :global(button[data-icon="add"]) {
          background-image: url(./images/AddButton.svg);
        }

        .Header :global(input) {
          background: var(--color-header-input-background);
          padding: 0 0.5rem;
          border: 0.6px solid var(--color-header-background);
          width: 20rem;
        }
        .Header :global(input:focus) {
          background: white;
        }
      `}</style>
    </>
  );
}
