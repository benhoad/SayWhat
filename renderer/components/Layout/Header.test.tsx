import { render } from "@testing-library/react";

import Header from "./Header";

describe("Header", () => {
  it("renders", () => {
    expect.hasAssertions();

    const { container } = render(<Header>HeaderContent</Header>);
    expect(container.firstElementChild).toMatchSnapshot();
  });
});
