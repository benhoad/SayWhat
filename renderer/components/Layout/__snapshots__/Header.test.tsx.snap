// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  class="jsx-505660841 Header"
>
  HeaderContent
</div>
`;
